ml-proto/src/spec/check.ml

```diff
@@ -179,14 +179,14 @@ let rec check_expr c ts e =
 
   | SetLocal (x, e1) ->
     check_expr c [local c x] e1;
-    check_type [] ts e.at
+    check_type [local c x] ts e.at
```

Comment:
I will, just need to read an introduction to ocaml somewhere, any good recommendation? (I already code in clojure, erlang and scheme so it can be a high level introduction)